src/main/java/org/example/gabartischolarsqbbe/service/QuestionService.java

```diff
@@ -0,0 +1,195 @@
+     * Process CSV file and save questions to database
+     * CSV format: question_text,difficulty,job_title_id,option1,option1_correct,option2,option2_correct,option3,option3_correct,option4,option4_correct
+     * @param file the CSV file to process
+     * @return CsvUploadResponse with processing results
+     */
+    public CsvUploadResponse processCsvFile(MultipartFile file) {
+        if (file.isEmpty()) {
+            return new CsvUploadResponse(0, 0, 0, "File is empty");
+        }
+        
+        if (!isValidCsvFile(file)) {
+            return new CsvUploadResponse(0, 0, 0, "Invalid file format. Please upload a CSV file");
+        }
+        
+        int totalProcessed = 0;
+        int successfullyInserted = 0;
+        int failed = 0;
+        StringBuilder errorMessages = new StringBuilder();
+        
+        try (CSVReader csvReader = new CSVReader(new InputStreamReader(file.getInputStream(), StandardCharsets.UTF_8))) {
+            List<String[]> records = csvReader.readAll();
+            
+            if (records.isEmpty()) {
+                return new CsvUploadResponse(0, 0, 0, "CSV file contains no records.");
+            }
+            // Skip header row if exists
+            boolean hasHeader = isHeaderRow(records.get(0));
+            int startIndex = hasHeader ? 1 : 0;
+            
+            for (int i = startIndex; i < records.size(); i++) {
+                String[] record = records.get(i);
+                totalProcessed++;
+                
+                try {
+                    Question question = parseRecordToQuestion(record);
+                    
+                    // Validate job title exists
+                    if (!jobService.jobExists(question.getJobTitleId())) {
+                        failed++;
+                        errorMessages.append("Row ").append(i + 1).append(": Job ID ")
+                                .append(question.getJobTitleId()).append(" does not exist. ");
+                        continue;
+                    }
+                    
+                    questionRepository.save(question);
+                    successfullyInserted++;
+                    
+                } catch (Exception e) {
+                    failed++;
+                    errorMessages.append("Row ").append(i + 1).append(": ").append(e.getMessage()).append(". ");
+                }
+            }
+            
+        } catch (IOException | CsvException e) {
+            return new CsvUploadResponse(0, 0, 0, "Error reading CSV file: " + e.getMessage());
+        }
+        
+        String message = String.format("Processing completed. %d successful, %d failed out of %d total records.", 
+                successfullyInserted, failed, totalProcessed);
+        
+        if (errorMessages.length() > 0) {
+            message += " Errors: " + errorMessages.toString();
+        }
+        
+        return new CsvUploadResponse(totalProcessed, successfullyInserted, failed, message);
+    }
```

Comment:
The method is heavily dense, try to break it into some logical chunks, and extract these chunks into methods